packages/datadog-plugin-aws-sdk/src/services/bedrockruntime/utils.js

```diff
@@ -350,14 +380,27 @@ function extractTextAndResponseReason (response, provider, modelName) {
         if (modelName.includes('embed')) {
           return new Generation({ message: body.embedding })
         }
-        const results = body.results || []
-        if (results.length > 0) {
-          const result = results[0]
+        if (body.results !== undefined) {
```

Comment:
similar to above, i think we can simplify
```suggestion
        if (body.results) {
```

Reply:
Making this change and also updating the condition right below to `else if (body.output)`